feat: fetch did file from canister metadata (#3558)
dfinity/sdk: CHANGELOG.md

# UNRELEASED

### feat: fetch did file from canister metadata when making canister calls

> `dfx canister call` will always fetch did file from the canister metadata. If the canister doesn't have the `candid:service` metadata, dfx will fallback to the current behavior of reading did file from the local build artifact. In a future release, we will deprecate this fallback option. This should not affect Motoko and Rust canisters built from dfx, as `dfx build` automatically write the Candid metadata into the canister.

Reviewer: ```suggestion
`dfx canister call` will always fetch the `.did` file from the canister metadata. If the canister doesn't have the `candid:service` metadata, dfx will fallback to the current behavior of reading the `.did` file from the local build artifact. This fallback behavior is deprecated and we will remove it in a future release. This should not affect Motoko and Rust canisters built from dfx, as `dfx build` automatically writes the Candid metadata into the canister.
```
